src/lib/util/dom.ts

```diff
@@ -61,3 +61,20 @@ export function parseDOM(html: string, baseUrl: string): Document {
 
     return doc;
 }
+
+// https://github.com/facebook/react/issues/10135#issuecomment-401496776
+// Via loujine's wikidata script.
+export function setInputValue(input: HTMLInputElement, value: string): void {
```

Comment:
Yeah I didn't want to do React workarounds too much, and it wouldn't really have been necessary either since we can call `onUrlBlur` and `onUrlChange` from the `MB` object, which I was trying to do first. However, those functions take a bunch of arguments, most of which we can compute ourselves quite easily, expect for `onUrlBlur`'s `canMerge`. Another reason why I decided not to go that route is that it can easily break, e.g. when the signatures of those functions are changed, there's nothing preventing us from sending the wrong data and possibly creating bad edits.

I guess the latter point also applies to our patches though, so I'll go ahead and make them a bit safer too.